src/main/java/org/mockbukkit/mockbukkit/inventory/meta/ShieldMetaMock.java

```diff
@@ -28,52 +38,256 @@ public ShieldMetaMock()
+	}
+
+	@Override
+	public int numberOfPatterns()
+	{
+		if (banner == null)
+		{
+			return 0;
+		}
+
+		return banner.numberOfPatterns();
 	}
 
 	@Override
 	public @Nullable DyeColor getBaseColor()
 	{
-		return this.color;
+		if (banner == null)
+		{
+			return null;
+		}
+
+		return banner.getBaseColor();
 	}
 
 	@Override
-	public void setBaseColor(@Nullable DyeColor color)
+	public void setBaseColor(@Nullable DyeColor baseColor)
 	{
-		this.color = color;
+		if (baseColor == null)
+		{
+			if (banner != null && banner.numberOfPatterns() > 0)
+			{
+				banner.setBaseColor(DyeColor.WHITE);
+			}
+			else
+			{
+				banner = null;
+			}
+		}
+		else
+		{
+			if (banner == null)
+			{
+				banner = new BannerStateMock(dyeColorToMaterial(baseColor));
+			}
+
+			banner.setBaseColor(baseColor);
+		}
 	}
 
 	@Override
 	protected void deserializeInternal(Map<String, Object> serialized)
 	{
 		super.deserializeInternal(serialized);
-		if (serialized.containsKey("base-color"))
+
+		String baseColor = (String) serialized.get("base-color");
+		if (baseColor != null)
+		{
+			banner = getBlockState(DyeColor.valueOf(baseColor));
+		}
+
+		Iterable<?> rawPatternList = (Iterable<?>) serialized.get("patterns");
+		if (rawPatternList == null)
+		{
+			return;
+		}
+
+		for (Object obj : rawPatternList)
 		{
-			this.color = DyeColor.values()[(int) serialized.get("base-color")];
+			Preconditions.checkArgument(obj instanceof Pattern, "Object (%s) in pattern list is not valid", obj.getClass());
+			addPattern((Pattern) obj);
 		}
 	}
 
 	@Override
 	public @NotNull Map<String, Object> serialize()
 	{
 		Map<String, Object> serialized = super.serialize();
-		if (getBaseColor() != null)
+		if (banner != null)
 		{
-			serialized.put("base-color", color.ordinal());
+			serialized.put("base-color", banner.getBaseColor().toString());
+
+			if (banner.numberOfPatterns() > 0)
+			{
+				serialized.put("patterns", banner.getPatterns());
+			}
 		}
+
 		return serialized;
 	}
 
-	public static ShieldMetaMock deserialize(Map<String, Object> serialized)
+	public static @NotNull ShieldMetaMock deserialize(@NotNull Map<String, Object> serialized)
 	{
 		ShieldMetaMock shieldMetaMock = new ShieldMetaMock();
 		shieldMetaMock.deserializeInternal(serialized);
 		return shieldMetaMock;
 	}
 
+	@Override
+	public boolean hasBlockState()
+	{
+		return banner != null;
+	}
+
+	@Override
+	public void clearBlockState()
+	{
+		banner = null;
+	}
+
+	@Override
+	public @NotNull BlockState getBlockState()
+	{
+		if (banner != null) return banner.copy();
+		return getBlockState(null);
+	}
+
+	@Override
+	public void setBlockState(@NotNull BlockState blockState)
+	{
+		Preconditions.checkArgument(blockState instanceof Banner, "Invalid blockState");
+		this.banner = (Banner) blockState;
+	}
+
+	private static Banner getBlockState(DyeColor color)
+	{
+		return new BannerStateMock(dyeColorToMaterial(color));
+	}
+
+	static Material dyeColorToMaterial(DyeColor color)
```

Comment:
Why is this package-private, and not just private?

Reply:
Made private.